### ISSUE TITLE
The `/sitemap/xsl/sitemap.xsl`  file is Not Found when I try to reach my sitemap `http://localhost:1337/sitemap/index.xml`

### ISSUE THREAD

**Issue by AliKdhim87** [boazpoolman/strapi-plugin-sitemap#97]
### Describe the bug

The problem is when I try to reach my sitemap through this URL http://localhost:1337/sitemap/index.xml, I got nothing rendered in the browser, and when I check the chrome dev tool network I got this error

```
Request URL: http://localhost:1337/sitemap/xsl/sitemap.xsl
Request Method: GET
Status Code: 404 Not Found
Remote Address: 127.0.0.1:1337
Referrer Policy: no-referrer

``` 

### Screenshots
![Screenshot 2022-10-24 at 12 50 49](https://user-images.githubusercontent.com/53454241/197514175-a74aafd8-2c73-4471-890a-93619a698009.png)

### System

- Node.js version:  `>=16.16.0 <=18.x.x`
- NPM version: `>=6.0.0`
- Strapi version: ` 4.4.1`
- Plugin version: `2.0.8`
- Database:
- Operating system:  macOS 

### Additional context

Maybe this is related to the issue when I installed the plugin I had to create a `sitemap` manually inside the `public` folder, otherwise, I get an error  

```
2022-10-24 13:25:54.947] error: [strapi-plugin-sitemap]: Something went wrong while trying to write the sitemap XML file to your public folder. Error: ENOENT: no such file or directory, open 'public/sitemap/index.xml'
Error
```

**Comment by pranavks** [comment]
+1

**Comment by boazpoolman** [comment]
Seems like a bug. I'll do some investigation asap.

**Comment by ArtKuz** [comment]
+1

**Comment by marcospassos** [comment]
Same problem here.

**Comment by boazpoolman** [comment]
So I did some investigation and concluded the following:

When Strapi is bootstrapping, the plugin should create a folder called `sitemap` in the `public` folder of your project.
In this folder there is a `.gitignore` file and a `xsl` folder with some xsl styling files.

If the creation of this `sitemap` folder failed and you try to generate the sitemap afterwards anyways you will get the error as OP described he got:

```
error: [strapi-plugin-sitemap]: Something went wrong while trying to write the sitemap XML file to your public folder. Error: ENOENT: no such file or directory, open 'public/sitemap/index.xml'
```

You could now create the `sitemap` folder manually, as OP did and generate the sitemap. You will now end up with the bug that the XSL file is missing as it was never correctly created.

**Temp fix:**
Copy the `xsl` folder [from here](https://github.com/boazpoolman/strapi-plugin-sitemap/tree/master/public) and paste it into the `public/sitemap` folder.

**Perm fix:**

**Comment by boazpoolman** [comment]
This has been fixed in [version 3.0.0](https://github.com/boazpoolman/strapi-plugin-sitemap/releases/tag/3.0.0) of the Sitemap plugin.

As of this new version, the plugin serves a "virtual sitemap". This is not a XML file stored in the public folder in your file system. Instead the XML gets saved to the database and an XML API endpoint is exposed to access the sitemap. 
This means that the `public/sitemap` folder has become redundant and the issue as described above cannot happend anymore.
